api/client/src/main/java/org/projectnessie/client/NessieConfigConstants.java

```diff
@@ -291,12 +291,140 @@ public final class NessieConfigConstants {
+      "nessie.authentication.oauth2.token-exchange.token-endpoint";
+
+  /**
+   * For token exchanges only. An alternate client ID to use. If not provided, the global client ID
+   * will be used. If provided, and if the client is confidential, then its secret must be provided
+   * as well with {@value #CONF_NESSIE_OAUTH2_TOKEN_EXCHANGE_CLIENT_SECRET} – the global client
+   * secret will NOT be used.
+   */
+  @ConfigItem(section = "OAuth2 Authentication")
+  public static final String CONF_NESSIE_OAUTH2_TOKEN_EXCHANGE_CLIENT_ID =
+      "nessie.authentication.oauth2.token-exchange.client-id";
+
+  /**
+   * For token exchanges only. The client secret to use, if {@value
+   * #CONF_NESSIE_OAUTH2_TOKEN_EXCHANGE_CLIENT_ID} is defined and the token exchange client is
+   * confidential.
+   */
+  @ConfigItem(section = "OAuth2 Authentication")
+  public static final String CONF_NESSIE_OAUTH2_TOKEN_EXCHANGE_CLIENT_SECRET =
+      "nessie.authentication.oauth2.token-exchange.client-secret";
+
+  /**
+   * For token exchanges only. A URI that indicates the target service or resource where the client
+   * intends to use the requested security token. Optional.
+   */
+  @ConfigItem(section = "OAuth2 Authentication")
+  public static final String CONF_NESSIE_OAUTH2_TOKEN_EXCHANGE_RESOURCE =
+      "nessie.authentication.oauth2.token-exchange.resource";
+
+  /**
+   * For token exchanges only. The logical name of the target service where the client intends to
+   * use the requested security token. This serves a purpose similar to the resource parameter but
+   * with the client providing a logical name for the target service.
+   */
+  @ConfigItem(section = "OAuth2 Authentication")
+  public static final String CONF_NESSIE_OAUTH2_TOKEN_EXCHANGE_AUDIENCE =
+      "nessie.authentication.oauth2.token-exchange.audience";
+
+  /**
+   * For token exchanges only. Space-separated list of scopes to include in each token exchange
+   * request to the OAuth2 server. Optional. If undefined, the global scopes configured through
+   * {@link #CONF_NESSIE_OAUTH2_CLIENT_SCOPES} will be used. If defined and null or empty, no scopes
+   * will be used.
+   *
+   * <p>The scope names will not be validated by the Nessie client; make sure they are valid
+   * according to <a href="https://datatracker.ietf.org/doc/html/rfc6749#section-3.3">RFC 6749
+   * Section 3.3</a>.
+   */
+  @ConfigItem(section = "OAuth2 Authentication")
+  public static final String CONF_NESSIE_OAUTH2_TOKEN_EXCHANGE_SCOPES =
+      "nessie.authentication.oauth2.token-exchange.scopes";
+
+  /**
+   * For token exchanges only. The subject token to exchange.
+   *
+   * <p>By default, the client will use its current access token as the subject token. But if this
+   * property is set, the client will use the static token provided here instead.
+   */
+  @ConfigItem(section = "OAuth2 Authentication")
+  public static final String CONF_NESSIE_OAUTH2_TOKEN_EXCHANGE_SUBJECT_TOKEN =
```

Comment:
Yes if you pass a token this way, it will be a static one. But if you don't pass anything, the client will use its current access token as the subject token – which is probably what most users will want to do.